src/network/websockets/serialization.ts

```diff
@@ -1,6 +1,6 @@
 export type Serialization = {
-  encode: (value: unknown) => string;
-  decode: (string: string) => unknown;
+  encode: (value: unknown, kind: any) => string;
```

Comment:
Could we future-proof the syntax? Let’s make the second parameter an object for possibilities to expand it in the future, and let’s include the id while we’re at it for symmetry.

```ts
export type Serialization = {
  encode: (value: unknown, details: { kind: string, id: string }) => string;
  decode: (string: string, details: { kind: string, id: string }) => unknown;
};
```

Reply:
done